test/e2e/framework/util.go

```diff
@@ -2405,10 +2405,17 @@ func WaitForStableCluster(c clientset.Interface, masterNodes sets.String) int {
 
 	}
 	allPods.Items = currentPods
+	systemPods, err := c.CoreV1().Pods(metav1.NamespaceSystem).List(metav1.ListOptions{})
```

Comment:
> allPods, err := c.CoreV1().Pods(metav1.NamespaceAll).List(metav1.ListOptions{})

this above is already problematic in case of thousands of pods.

Reply:
@neolit123 could you clarify how I could fix this? It seems this function is already dependent on polling all of the pods in the cluster. I could pull the system pods from that query above, but the system pod query is likely to usually be pretty small, right?

Reply:
@neolit123  doesn't the list of pods already exist in the informer cache? also the scheduler tests which uses this function does not really create thousands of pods.

Reply:
the system pods should be fine, but "all the pods" can block for a while in super-clusters and i wouldn't be surprised that nobody in the wild would run our test suite on such clusters because of that (possibly other reasons too). it feels to me this logic should reworked.

Reply:
@ahg-g the cache can help.
ok so this function is only used under test/e2e/scheduling/predicates.go
this means that this function has to be moved under `test/e2e/scheduling/` ideally.

i was under the impression that given this is part of the framework we are using it elsewhere which doesn't make sense.

Reply:
cc @oomichi for an opinion about the move.

Reply:
yes, this seems to be used only in the scheduler e2e tests, I agree it should be moved under ```test/e2e/scheduling```